core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -117,11 +116,16 @@ public void commit(TableMetadata base, TableMetadata metadata) {
         !metadata.properties().containsKey(TableProperties.WRITE_METADATA_LOCATION),
         "Hadoop path-based tables cannot relocate metadata");
 
-    Path tempMetadataFile = metadataPath(UUID.randomUUID().toString() + TableMetadataParser.getFileExtension(conf));
+    boolean metadataCompressionEnabled = metadata.propertyAsBoolean(
+        TableProperties.METADATA_COMPRESSION_ENABLED,
+        TableProperties.METADATA_COMPRESSION_ENABLED_DEFAULT);
+
+    String fileExtension = TableMetadataParser.getFileExtension(metadataCompressionEnabled);
+    Path tempMetadataFile = metadataPath(UUID.randomUUID().toString() + fileExtension);
```

Comment:
Does the temp file need the extension? May be simpler to leave it out.

Reply:
I kept it consistent with what we had before but it will simplify the code if we leave it out. +1

Reply:
Either way.

Reply:
I kept it because the temporary file will be written without compression otherwise.
`TableMetadataParser` inspects file extensions.